spring-integration-r2dbc/src/test/java/org/springframework/integration/r2dbc/inbound/R2dbcMessageSourceTests.java

```diff
@@ -129,6 +135,135 @@ public void validateSuccessfulQueryWithMultipleElementOfFluxDBObject() {
 
 	}
 
+	@Test
+	public void validateSuccessfulUpdateWithSingleElementOfMonoDBObject() {
+		this.entityTemplate.insert(new Person("Bob", 35))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		r2dbcMessageSourceSelectMany.setUpdateSql("UPDATE Person SET name='Foo' where age = :age");
+		r2dbcMessageSourceSelectMany.setBindFunction((DatabaseClient.BindSpec<?> bindSpec, Person o) -> bindSpec.bind(
+				"age", o.getAge()));
+		r2dbcMessageSourceSelectMany.setExpectSingleResult(true);
+
+		StepVerifier.create(r2dbcMessageSourceSelectMany.receive().getPayload())
+				.assertNext(person -> assertThat(((Person) person).getName()).isEqualTo("Bob"))
+				.verifyComplete();
+
+		this.entityTemplate.select(Person.class)
+				.all()
+				.as(StepVerifier::create)
+				.assertNext(person -> assertThat(person.getName()).isEqualTo("Foo"))
+				.verifyComplete();
+
+	}
+
+	@Test
+	public void validateSuccessfulUpdateWithMultiplesElementsOfFluxDBObject() {
+		this.entityTemplate.insert(new Person("Bob", 35))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		this.entityTemplate.insert(new Person("Tom", 40))
+				.then()
+				.as(StepVerifier::create)
+				.verifyComplete();
+
+		r2dbcMessageSourceSelectMany.setUpdateSql("UPDATE person SET name='Foo' where id = :id");
+		r2dbcMessageSourceSelectMany.setBindFunction((DatabaseClient.BindSpec<?> bindSpec, Person o) -> bindSpec.bind(
+				"id",	o.getId()));
+		StepVerifier.create(r2dbcMessageSourceSelectMany.receive().getPayload())
+				.expectNextCount(2)
+				.verifyComplete();
+
+		this.entityTemplate.select(Person.class)
+				.all()
+				.as(StepVerifier::create)
+				.assertNext(person -> assertThat(person.getName()).isEqualTo("Foo"))
+				.assertNext(person -> assertThat(person.getName()).isEqualTo("Foo"))
```

Comment:
May we be sure here that these persons are really different?
Maybe checking their age...
Right now their names are the same, so it isn't clear if both asserts are not against the same record.